Reject letter input without row number

EricccTaiwan opened the pull request:
This patch improves input validation by ensuring that a letter input must be followed by a row number. If the user enters only a letter within the valid board range without a corresponding row number, an error message is displayed. This enhances clarity and prevents invalid moves, ensuring correct input handling.

Before
```shell
$ ./ttt
 1 |             
 2 |             
 3 |             
 4 |             
---+-------------
      A  B  C  D
X> 1
Invalid operation: No leading alphabet
X> 2
Invalid operation: No leading alphabet
X> a
X> b
X> c
X> d
X> e
Invalid operation: index exceeds board size
X> ^C
$
```
After
```shell
$ ./ttt
 1 |             
 2 |             
 3 |             
 4 |             
---+-------------
      A  B  C  D
X> 1
Invalid operation: No leading alphabet
X> 2
Invalid operation: No leading alphabet
X> a
Invalid operation: No row number
X> b
Invalid operation: No row number
X> c
Invalid operation: No row number
X> d
Invalid operation: No row number
X> e
Invalid operation: index exceeds board size
X> f
Invalid operation: index exceeds board size
X> ^C
$
```

jserv reviewed (changes requested):
Instead of "trailing number," use "row numbers" for precise naming.

jserv commented:
The term "row numbers" is clearer and more standard, ensuring that the input includes both the row (number) and the column (alphabet).

EricccTaiwan commented:
Thanks for your feedback! "row numbers" sounds clearer, but since only one row number is entered at a time, should it be singular ("row number") instead?

jserv commented:
> Thanks for your feedback! "row numbers" sounds clearer, but since only one row number is entered at a time, should it be singular ("row number") instead?

The term "row numbers" means the collection of the numbers in row, and you should use the form "row number" for each input.

EricccTaiwan committed:
Reject letter input without row number

This patch improves input validation by ensuring that a letter input
must be followed by a row number. If the user enters only a letter
within the valid board range without a corresponding row number, an
error message is displayed. This enhances clarity and prevents invalid
moves, ensuring correct input handling.

Co-authored-by: charliechiou <charlie910417@gmail.com>

jserv commented:
Thank @EricccTaiwan for contributing!
